### DISCUSSION TITLE
HTML Sanitizer

### DISCUSSION THREAD

**Discussion by tonysm** [laravel/framework#42827] [Ideas]
Looks like Symfony has a stable release of their [HTML Sanitizer](https://symfony.com/doc/current/html_sanitizer.html). I was wondering if it wouldn't be cool to have something like that upstream in the Laravel core. Maybe something like:

```blade
{{ s($post->body) }}
```

Assuming the body field there has HTML on it, the `s()` function would _sanitize_ the content and return an instance of `HtmlString`.

It could ship only with the [Element Baseline](https://symfony.com/doc/current/html_sanitizer.html#allow-element-baselines) and allow app-land code to override it using a callable, something like:

```php
use Illuminate\View\HtmlSanitizer;
use Symfony\Component\HtmlSanitizer\HtmlSanitizer as SymfonyHtmlSanitizer;
use Symfony\Component\HtmlSanitizer\HtmlSanitizerConfig;

// Somewhere in AppServiceProvider
HtmlSanitizer::configureUsing(fn () => new SymfonyHtmlSanitizer(
    (new HtmlSanitizerConfig())
        ->blockElement('h1')
));
```

This being in the core would mean more eyes on how sanitization is done.

**Comment by jva91** [comment]
This would be really nice when a sanitizer option is available in Laravel core. The Symfony Sanitizer looks nice to implement
